symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV2.java

```diff
@@ -1,36 +1,196 @@
+ * If the retrieved list is empty, the BDK bot will create a new datafeed to listen.
+ *
+ * The BDK bot will listen to this datafeed to get all the received real-time events.
+ *
+ * If this datafeed becomes stale or faulty, the BDK bot will create the new one for listening.
+ *
+ * This service will be stopped by calling {@link DatafeedServiceV2#stop()}
+ *
+ * If the datafeed service is stopped during a read datafeed call, it has to wait until the last read finish to be
+ * really stopped
  */
+@Slf4j
 public class DatafeedServiceV2 extends AbstractDatafeedService {
 
     private final AtomicBoolean started = new AtomicBoolean();
+    private final AckId ackId;
+    private V5Datafeed datafeed;
 
-    public DatafeedServiceV2(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
-        super(agentClient, podClient, authSession, config);
+    public DatafeedServiceV2(ApiClient agentClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, authSession, config);
+        this.ackId = new AckId().ackId("");
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void start() {
+    public void start() throws ApiException, AuthUnauthorizedException {
+        if (this.started.get()) {
+            throw new IllegalStateException("The datafeed service is already started");
+        }
+        try {
+            this.datafeed = this.retrieveDatafeed();
+            if (this.datafeed == null) {
+                this.datafeed = this.createDatafeed();
+            }
+            log.debug("Start reading datafeed events");
+            do {
+                this.started.set(true);
+                this.readDatafeed();
+            } while (this.started.get());
+        } catch (AuthUnauthorizedException | ApiException exception) {
+            throw exception;
+        } catch (Throwable throwable) {
+            log.error("Unknown error", throwable);
+        }
+    }
 
+    protected AckId getAckId() {
+        return this.ackId;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void stop() {
+        this.started.set(false);
+    }
+
+    private V5Datafeed createDatafeed() throws Throwable {
+        log.debug("Start creating datafeed from agent");
+        Retry retry = this.getRetryInstance("Create V5Datafeed");
+        return retry.executeCheckedSupplier(() -> {
+            try {
+                return this.datafeedApi.createDatafeed(authSession.getSessionToken(), authSession.getKeyManagerToken());
+            } catch (ApiException e) {
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                } else {
+                    log.error("Error {}: {}", e.getCode(), e.getMessage());
+                }
+                throw e;
```

Comment:
If `e.isUnauthorized()` it means we will execute `authSession.refresh()` and throw the error. Is this intended ?